Doc/library/asyncio-sync.rst

```diff
@@ -1,305 +1,324 @@
 .. currentmodule:: asyncio
+
 .. _asyncio-sync:
 
-Synchronization primitives
 ==========================
+Synchronization Primitives
+==========================
+
+asyncio synchronization primitives are designed to be similar to
+those of the :mod:`threading` module with two important caveats:
+
+* asyncio primitives are not thread-safe, therefore they should not
+  be used for OS threads synchronization (use :mod:`threading` for
+  that);
 
-**Source code:** :source:`Lib/asyncio/locks.py`
+* methods of synchronization objects do not accept the *timeout*
+  argument; use the :func:`asyncio.wait_for` function to perform
+  operations with timeouts.
 
-Locks:
+asyncio has the following basic primitives:
 
 * :class:`Lock`
 * :class:`Event`
 * :class:`Condition`
-
-Semaphores:
-
 * :class:`Semaphore`
 * :class:`BoundedSemaphore`
 
-asyncio lock API was designed to be close to classes of the :mod:`threading`
-module (:class:`~threading.Lock`, :class:`~threading.Event`,
-:class:`~threading.Condition`, :class:`~threading.Semaphore`,
-:class:`~threading.BoundedSemaphore`), but it has no *timeout* parameter. The
-:func:`asyncio.wait_for` function can be used to cancel a task after a timeout.
+
+---------
 
 
 Lock
-----
+====
 
 .. class:: Lock(\*, loop=None)
 
-   Primitive lock objects.
+   Implements a mutex lock for asyncio tasks.  Not thread-safe.
 
-   A primitive lock is a synchronization primitive that is not owned by a
-   particular coroutine when locked.  A primitive lock is in one of two states,
-   'locked' or 'unlocked'.
+   An asyncio lock can be used to guarantee exclusive access to a
+   shared resource to a single asyncio Task.
```

Comment:
~to~ of

Reply:
"of a single asyncio Task."?  We use locks to guarantee access *(access to a shared resource)* to an asyncio task.  With "of" I'm not sure it reads correctly...

Reply:
Maybe: "to a single asyncio Task and its shared resource.

Reply:
The problem is that the resource (typically) does not belong to the Task -- the Task just wants to exclusively work with the resource for some time; hence "**its** shared resource." isn't 100% correct.

Maybe we should just drop the "to a single asyncio Task" part:

```
An asyncio lock can be used to guarantee exclusive access to a shared resource.
```

?

Reply:
I like that. Dropping the phrase is better.